climakitae/util/generate_gwl_timing_table.py

```diff
@@ -0,0 +1,95 @@
+""" 
+Util for generating estimated crossing years for global warming levels
+
+Outputs a csv file for reference in ../data/ ###
+
+To run, type: <<python generate_gwl_timing_table.py>> in the command line. Runs in about 20 seconds and returns no output.
+"""
+
+from climakitae.util.utils import read_csv_file, write_csv_file
+import numpy as np
+import pandas as pd
+from climakitae.core.paths import (
+    ssp119_file,
+    ssp126_file,
+    ssp245_file,
+    ssp370_file,
+    ssp585_file,
+)
+
+
+def main():
+    """
+    Generates global warming level (GWL) timing table based on the IPCC warming trajectories and ranges in ../data/
+
+    Uses the same method that  ../explore/warming.py uses to generate visualizations to identify crossing years.
+
+    Saves the mean, 5th, and 95th percentile for each SSP scenario
+
+    """
+
+    # load IPCC scenaario trajectories
+    ssp119_data = read_csv_file(ssp119_file, index_col="Year")
+    ssp126_data = read_csv_file(ssp126_file, index_col="Year")
+    ssp245_data = read_csv_file(ssp245_file, index_col="Year")
+    ssp370_data = read_csv_file(ssp370_file, index_col="Year")
+    ssp585_data = read_csv_file(ssp585_file, index_col="Year")
+
+    ssp_dict = {
+        # "SSP 1-1.9": ssp119_data, # skip this one scenario because it does not reach any warming levels
+        "SSP 1-2.6": ssp126_data,
+        "SSP 2-4.5": ssp245_data,
+        "SSP 3-7.0": ssp370_data,
+        "SSP 5-8.5": ssp585_data,
```

Comment:
slightly inclined to not have spaces in column names; i.e. "SSP_1-2.6" (replace space with underscore) but am open to being disagreed with here

Reply:
Changed in latest push